scrape/scrape.go

```diff
@@ -413,23 +421,44 @@ func (sp *scrapePool) Sync(tgs []*targetgroup.Group) {
 	start := time.Now()
 
 	sp.targetMtx.Lock()
-	var all []*Target
-	sp.droppedTargets = []*Target{}
+	
+	targetCount := 0
```

Comment:
If you're not already iterating through the target groups, there's not too much point pre-allocating the size of the slice.

Reply:
Your prev comment was about adding cap here :)